### PR TITLE
Use auth from opsrc

### PR DESCRIPTION
**Pull request by kevinrizza**
-Query opsrc for secret name
-Update downloader to get auth token secret
-Use auth token in apprclient request header

### CODE REVIEW

**Comment by kevinrizza** [comment]
Put this wip pr up for now to review, still need to merge the auth pr on operator-marketplace and then update vendored dependencies.

File: pkg/appregistry/downloader.go [hunk 1]
```diff
@@ -178,3 +198,24 @@ func (d *downloader) QuerySource(key *types.NamespacedName) (spec *v1alpha1.Oper
...
+	}
+
+	auth := spec.AuthorizationToken
+	if auth.SecretName != "" {
+		secret, err := d.secretClient.CoreV1().Secrets(namespace).Get(auth.SecretName, metav1.GetOptions{})
+		if err != nil {
+			return options, err
+		}
+
+		options.AuthToken = string(secret.Data["token"])
```
Comment (tkashem) [inline pkg/appregistry/downloader.go]: I thought we needed to base64 decode it.

File: pkg/appregistry/appregistry.go [hunk 2]
```diff
@@ -100,3 +107,23 @@ func NewClient(kubeconfig string, logger *logrus.Entry) (clientset marketplace.I
 	clientset, err = marketplace.NewForConfig(config)
 	return
 }
+
+func NewSecretClient(kubeconfig string, logger *logrus.Entry) (clientset *kubernetes.Clientset, err error) {
```
Comment (tkashem) [inline pkg/appregistry/appregistry.go]: nit: maybe `kubeclient` is a more appropriate name?

File: pkg/appregistry/downloader.go (hunk 1 above)
Comment (kevinrizza) [reply pkg/appregistry/downloader.go]: @tkashem Nope, the api just obfuscates it as base64, but when pulling from the client directly it is returned as a string.

File: pkg/appregistry/appregistry.go (hunk 2 above)
Comment (kevinrizza) [reply pkg/appregistry/appregistry.go]: Sure, updated in latest commit. I also updated the field name in the downloader to match

**Pull request merged**